src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -197,109 +507,213 @@
+                        configLog.log(System.Logger.Level.ERROR,
+                                "FilterFactory provided on the command line can not be overridden");
+                        // Do not continue if configuration not initialized
+                        throw new ExceptionInInitializerError("FilterFactory provided on the command line can not be overridden");
+                    }
+                    factory = f;
+                } catch (RuntimeException | ClassNotFoundException | NoSuchMethodException |
+                        IllegalAccessException | InstantiationException | InvocationTargetException ex) {
+                    configLog.log(System.Logger.Level.ERROR,
+                            "Error configuring filter factory", ex);
+                    // Do not continue if configuration not initialized
+                    throw new ExceptionInInitializerError(
+                            "FilterFactory configuration: jdk.serialFilterFactory: " + ex.getMessage());
+                }
+            } else {
+                factory = new BuiltinFilterFactory();
+            }
+            serialFilterFactory = factory;
 
             // Setup shared secrets for RegistryImpl to use.
             SharedSecrets.setJavaObjectInputFilterAccess(Config::createFilter2);
         }
 
         /**
-         * Current configured filter.
+         * Config has no instances.
          */
-        private static volatile ObjectInputFilter serialFilter = configuredFilter;
+        private Config() {
+        }
 
         /**
-         * Returns the system-wide serialization filter or {@code null} if not configured.
+         * Logger for filter actions.
+         */
+        private static void traceFilter(String msg, Object... args) {
+            if (traceFilters && configLog != null) {
+                configLog.log(TRACE, msg, args);
+            }
+        }
+
+        /**
+         * Returns the static JVM-wide deserialization filter or {@code null} if not configured.
          *
-         * @return the system-wide serialization filter or {@code null} if not configured
+         * @return the static JVM-wide deserialization filter or {@code null} if not configured
          */
         public static ObjectInputFilter getSerialFilter() {
             return serialFilter;
         }
 
         /**
-         * Set the system-wide filter if it has not already been configured or set.
+         * Set the static JVM-wide filter if it has not already been configured or set.
          *
-         * @param filter the serialization filter to set as the system-wide filter; not null
+         * @param filter the deserialization filter to set as the JVM-wide filter; not null
          * @throws SecurityException if there is security manager and the
          *       {@code SerializablePermission("serialFilter")} is not granted
          * @throws IllegalStateException if the filter has already been set {@code non-null}
```

Comment:
`* @throws IllegalStateException if the filter has already been set {@code non-null}`

Is there a typo/word missing ?

Reply:
non-null is unnecessary.